From 910a14205035a32d079fb76efe95edd229686626 Mon Sep 17 00:00:00 2001
From: Hien Dang <hien.dang.eb@rvc.renesas.com>
Date: Thu, 2 Jun 2016 10:35:26 +0700
Subject: [PATCH 157/909] soc: renesas: rcar_ems_ctrl: Exclude A53 out of
 targeted frequency scaled CPU

commit 79b4ea74d1a827a202190d72d667ef17786f2c85 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Currently, CPUFreq is just enabled for CA57 only.
EMS cannot change the CA53 cores' frequency,
so CA53 cores should be excluded from
EMS targeted frequency scaled CPU.

This patch is implemented for that.

Signed-off-by: Hien Dang <hien.dang.eb@rvc.renesas.com>
Signed-off-by: Dien Pham <dien.pham.ry@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/renesas/rcar_ems_ctrl.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/renesas/rcar_ems_ctrl.c b/drivers/soc/renesas/rcar_ems_ctrl.c
index cafca878642f..8295365075c5 100644
--- a/drivers/soc/renesas/rcar_ems_ctrl.c
+++ b/drivers/soc/renesas/rcar_ems_ctrl.c
@@ -247,7 +247,7 @@ static struct notifier_block ems_cpufreq_notifier_block = {
 static int __init rcar_ems_cpu_shutdown_init(void)
 {
 	int cpu;
-	struct device_node *cpu_node, *ems_node;
+	struct device_node *cpu_node, *ems_node, *tmp_node;
 	int total_target_cpu, i;
 
 	if (!IS_ENABLED(CONFIG_RCAR_THERMAL_EMS_ENABLED))
@@ -265,9 +265,12 @@ static int __init rcar_ems_cpu_shutdown_init(void)
 						"target_cpus", 0);
 
 	for_each_online_cpu(cpu) {
+		tmp_node  = of_get_cpu_node(cpu, NULL);
+		if (!of_device_is_compatible(tmp_node, "arm,cortex-a57"))
+			continue;
 		for (i = 0; i < total_target_cpu; i++) {
 			cpu_node = of_parse_phandle(ems_node, "target_cpus", i);
-			if (of_get_cpu_node(cpu, NULL) == cpu_node) {
+			if (tmp_node == cpu_node) {
 				cpumask_set_cpu(cpu, &target_cpus);
 				break;
 			}
-- 
2.17.1

